src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/AnonymousTypeDescriptor.cs

```diff
@@ -79,7 +79,7 @@ internal bool Equals(AnonymousTypeDescriptor other, bool ignoreCustomModifiersAn
             ImmutableArray<AnonymousTypeField> otherFields = other.Fields;
             for (int i = 0; i < count; i++)
             {
-                if (!myFields[i].Type.Equals(otherFields[i].Type, ignoreCustomModifiersAndArraySizesAndLowerBounds, ignoreDynamic))
+                if (!myFields[i].Type.Equals(otherFields[i].Type, comparison & ~TypeCompareKind.IgnoreTupleNames))
```

Comment:
It would be interesting to come up with a testcase for this. This is about equivalence of anonymous types when they contain tuple members.